### PR TITLE
Fixes #23377 - audit external group membership changes

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

**Review by ares — CHANGES_REQUESTED** [review]
tests are being worked on by @kgaikwad

File: app/models/user.rb [hunk 1]
```diff
@@ -314,6 +314,10 @@ def set_lower_login
     self.lower_login = login.downcase if login.present?
   end
 
+  def self.fetch_ids_by_list(userlist)
+    User.where(:lower_login => userlist.map(&:downcase)).pluck(:id)
```
Comment (ares) [inline app/models/user.rb:318]: could you please add a test for this method that would ensure it's case insensitive? basically to replace tests for add_users/remote_users this PR is dropping, it should verify that even if we pass OnE, TWO it affects users one, two

**Review by ares — CHANGES_REQUESTED** [review]
Works well, I have only one suggestion for keeping some test logic. If that's kept and tests are green, it's ready for merge.

**Comment by kgaikwad** [comment]
@ares,
Updated PR with a test case.

File: test/models/user_test.rb [hunk 2]
```diff
@@ -1232,4 +1232,20 @@ def editing_self_helper
...
+
+    usergroup = FactoryBot.create(:usergroup)
+    usergroup.user_ids = user_ids
+    usergroup.save
+
+    login_values = usergroup.users.map(&:login).sort
+
+    # users 'one' 'two' are defined in fixtures, 'three' is not defined
+    assert_equal ['one', 'two'], login_values
+    assert (!login_values.include? 'tHREE')
```
Comment (ares) [inline test/models/user_test.rb]: better written as [refute_includes](https://apidock.com/ruby/v1_9_3_125/MiniTest/Assertions/refute_includes)

File: test/models/user_test.rb [hunk 3]
```diff
@@ -1232,4 +1232,20 @@ def editing_self_helper
       assert_nil recent_audit.audited_changes[:roles]
     end
   end
+
+  test '#fetch_ids_by_list is case insensitive' do
+    user_ids = User.fetch_ids_by_list(['OnE', 'TwO', 'tHREE'])
+    assert_not_empty user_ids
```
Comment (ares) [inline test/models/user_test.rb]: I think this is covered by the line below

**Review by ares — CHANGES_REQUESTED** [review]
two small comments to the test, good to be merged afterwards

**Comment by kgaikwad** [comment]
@ares, 
Updated PR as per review comments.
Some tests are failing but those are I guess irrelevant to my changes.

**Review by ares — APPROVED** [review]

**Comment by ares** [comment]
Test failures are caused by develop branch, it was already fixed so your PR would have to be rebased. But I see no other tests are failing, so let's merge this. Thanks @kgaikwad!

**Pull request merged**